python_modules/dagster/dagster/_core/definitions/asset_graph_subset.py

```diff
@@ -1,70 +1,116 @@
 import operator
 from collections import defaultdict
 from datetime import datetime
-from typing import AbstractSet, Any, Callable, Dict, Iterable, Mapping, Optional, Set, Union, cast
+from typing import (
+    AbstractSet,
+    Any,
+    Callable,
+    Dict,
+    Iterable,
+    Mapping,
+    NamedTuple,
+    Optional,
+    Set,
+    Union,
+    cast,
+)
 
 from dagster import _check as check
 from dagster._core.definitions.partition import (
     PartitionsDefinition,
     PartitionsSubset,
 )
+from dagster._core.definitions.time_window_partitions import PartitionKeysTimeWindowPartitionsSubset
 from dagster._core.errors import (
     DagsterDefinitionChangedDeserializationError,
 )
 from dagster._core.instance import DynamicPartitionsStore
+from dagster._serdes.serdes import (
+    NamedTupleSerializer,
+    SerializableNonScalarKeyMapping,
+    whitelist_for_serdes,
+)
 
 from .asset_graph import AssetGraph
 from .events import AssetKey, AssetKeyPartitionKey
 
 
-class AssetGraphSubset:
-    def __init__(
-        self,
-        asset_graph: AssetGraph,
-        partitions_subsets_by_asset_key: Optional[Mapping[AssetKey, PartitionsSubset]] = None,
-        non_partitioned_asset_keys: Optional[AbstractSet[AssetKey]] = None,
-    ):
-        self._asset_graph = asset_graph
-        self._partitions_subsets_by_asset_key = partitions_subsets_by_asset_key or {}
-        self._non_partitioned_asset_keys = non_partitioned_asset_keys or set()
+class PartitionsSubsetMappingNamedTupleSerializer(NamedTupleSerializer):
```

Comment:
@sryza I've refactored this serializer to be generalizable to other named tuples that contain partitions subsets mappings. Unfortunately serializers  are instantiated on demand rather than at definition time, so we can't easily provide field names to apply this custom logic.

Instead, I've added logic to detect when partitions subsets exist and to convert those accordingly

Reply:
That makes sense

Reply:
This could use short docstring.